feat(frontend): implement question service

songgthu opened the pull request:
# Description

This PR implements the frontend for the Question Service and integrates question service backend, accessible only to admin users.

# Features
### 1. Question Bank Management:
- View the complete list of questions
- Filter and sort questions by topic, difficulty, and status
### 2. CRUD Operations:
- Add new questions
- Edit existing questions
- Archive or restore questions
### 3. Detailed Question View:
- View specific question details, including description and examples
- View suggested solutions

# Screenshots
## Admin homepage

<img width="1895" height="961" alt="image" src="https://github.com/user-attachments/assets/174c69a2-36a4-48be-ba7b-0705fa019315" />

## Admin question dashboard

<img width="987" height="722" alt="Screenshot 2025-11-07 at 11 32 46 AM" src="https://github.com/user-attachments/assets/1b2c2f0a-b3f8-4855-bc17-ee089bd6443d" />

## Admin question view

<img width="994" height="737" alt="Screenshot 2025-11-07 at 11 33 04 AM" src="https://github.com/user-attachments/assets/26ab15eb-6723-4f38-9c61-e73808629859" />

## Admin question edit

<img width="989" height="730" alt="Screenshot 2025-11-07 at 11 35 13 AM" src="https://github.com/user-attachments/assets/012fbd76-94fd-4924-888b-e49809cc27a3" />

## Admin question create

<img width="991" height="738" alt="Screenshot 2025-11-07 at 11 32 56 AM" src="https://github.com/user-attachments/assets/530ac071-52c8-4b16-85e7-f40ad37d8cb0" />

songgthu committed:
add question page and forms to create, edit, and view questions

songgthu committed:
Add dummy functions to integrate with question service backend

songgthu committed:
merged from dev

songgthu committed:
Fix bug in public-route to redirect based on user role

ncduy0303 commented on frontend/src/app/question/[id]/page.tsx:
`QuestionViewPage` is public route which is okay for viewing the question, but might not be okay for editting the question (i.e., only logged-in admin should be able to edit). Might need to modify the `QuestionCard` component.

songgthu replied:
Actually non-admin users should not allowed to access QuestionViewPage also, as question bank is only accessible to admin. I have updated the page with ProtectedRoute

songgthu committed:
Add protected route for QuestionViewPage

songgthu committed:
update pnpm-lock.yaml

songgthu committed:
merged from dev

songgthu committed:
Add function to auto-create admin user

DanielJames0302 commented on frontend/src/components/question/question-add-form.tsx:
```diff
@@ -0,0 +1,185 @@
+                  onChange={(e) => setQuestion({ ...question, topic: e.target.value })}
+                  required
+                />
+              </div>
```
- To ensure consistent topic naming, fetch the existing list of topics from Question Service API and replace the input field with a dropdown populated by these topics. The dropdown should also allow adding a new topic if it doesn’t already exist.

- When a new topic is added, automatically format it so that each word in the topic name starts with a capital letter.

- Implement validation to prevent duplicate topics—if the entered topic name already exists, the system should block the creation of that duplicate.

- Do the same for Question Edit page

songgthu committed:
Add dropdown menu for topic selection

songgthu committed:
update pnpm-lock

songgthu committed:
Merge dev into branch

ho-thanh-und committed:
update docker-compose to include health checks for mongodb and kafka

ho-thanh-und committed:
update question schema and functionalities

ho-thanh-und committed:
add solution schema and seeding

songgthu committed:
merged from dev

songgthu committed:
wip: integrate question service backend

ho-thanh-und committed:
update schema and controllers to faciliate question status

ho-thanh-und committed:
update seed for question and solution

ho-thanh-und committed:
update schema to allow solution fetching for corresponding question and programming language

songgthu committed:
update frontend integration to match backend schema

songgthu committed:
fix eslint warning

songgthu committed:
add archived questions view and integrate restore question endpoint

ho-thanh-und committed:
fix solution fetching logic and sanitise

ho-thanh-und committed:
fix restore functionality bug

DanielJames0302 committed:
Merge branch 'dev' into frontend-question-service

DanielJames0302 committed:
refactor: update MongoDB URI environment variable and enhance question details retrieval

- Changed environment variable from MONGODB_URI to MONGO_URI in docker-compose.yml.
- Updated fetchQuestionDetails method in RoomController to include archived questions in the API request.
- Modified frontend components to display question status, including an "Archived" badge where applicable.
- Updated API client to support fetching archived questions.

DanielJames0302 committed:
chore: update pnpm-lock.yaml

ho-thanh-und committed:
add .dockerignore

DanielJames0302 committed:
chore: update docker-compose.yml to add environment variables and service dependencies

ho-thanh-und committed:
update auto-seeding functionality for question solutions

DanielJames0302 committed:
Merge branch 'frontend-question-service' of https://github.com/CS3219-AY2526Sem1/cs3219-ay2526s1-project-g13 into frontend-question-service

DanielJames0302 reviewed (approved):
LGTM
